aixplain/modules/finetune.py

```diff
@@ -0,0 +1,134 @@
+__author__ = "lucaspavanelli"
+
+"""
+Copyright 2022 The aiXplain SDK authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Author: Duraikrishna Selvaraju, Thiago Castro Ferreira, Shreyas Sharma and Lucas Pavanelli
+Date: June 14th 2023
+Description:
+    FineTune Class
+"""
+from typing import List, Text
+import logging
+from aixplain.utils.file_utils import _request_with_retry
+import json
+from urllib.parse import urljoin
+from aixplain.utils import config
+from aixplain.factories.model_factory import ModelFactory
+from aixplain.modules.asset import Asset
+from aixplain.modules.dataset import Dataset
+from aixplain.modules.model import Model
+from aixplain.modules.finetune_cost import FinetuneCost
+from aixplain.decorators.api_key_checker import check_api_key
+
+
+class Finetune(Asset):
+    """FineTune is a powerful tool for fine-tuning machine learning models and using your own datasets for specific tasks.
+
+    Attributes:
+        name (Text): Name of the FineTune.
+        dataset_list (List[Dataset]): List of Datasets to be used for fine-tuning.
```

Comment:
maybe change the name for `datasets`

Reply:
@thiago-aixplain, here I am following the same convention as we have in Benchmark: https://github.com/aixplain/aiXplain/blob/main/aixplain/factories/benchmark_factory.py#L194